build-system/tasks/runtime-test/runtime-test-base.js

```diff
@@ -168,10 +167,10 @@ function updateReporters(config) {
     config.reporters.push('coverage-istanbul');
   }
 
-  if (isTravisPushBuild() && JSON_REPORT_TEST_TYPES.has(config.testType)) {
+  if (argv.report) {
     config.reporters.push('json-result');
     config.jsonResultReporter = {
-      outputFile: `results_${config.testType}.json`,
+      outputFile: resultFilename(config.testType),
```

Comment:
Would rather see `result-reports/${testType}.json` here instead of exporting a function from the report upload task. The function is simple enough, and this helps keep the gulp task modular.

Reply:
+1

Reply:
My one concern was that if someone wants to change the name of the output file, they'll have to change it in several places. I agree, though, that modularity is more important in this case -- I'll remove it.

Reply:
Is this function used anywhere else that would need to be changed?

Reply:
Not at the moment; if we change our mind we can add it back later

Reply:
Done removing it.